src/NServiceBus.RabbitMQ/Routing/DirectRoutingTopology.cs

```diff
@@ -18,6 +19,12 @@ public DirectRoutingTopology(Conventions conventions, bool useDurableExchanges)
 
         public void SetupSubscription(IModel channel, Type type, string subscriberName)
         {
+            if (type.IsInterface)
```

Comment:
Instead of adding this check here, I think it should go into `DefaultRoutingKeyConvention.GetRoutingKey` instead, where we can use the `interfaces` collection to log a warning only if there are problematic interfaces found.

This also lets us simplify the warning message by not needing to mention non-system messages.